metrics-facade-base/src/main/java/com/ringcentral/platform/metrics/instrument/executors/MonitoredExecutorService.java

```diff
@@ -0,0 +1,89 @@
+package com.ringcentral.platform.metrics.instrument.executors;
+
+import com.ringcentral.platform.metrics.MetricKey;
+import com.ringcentral.platform.metrics.MetricRegistry;
+import com.ringcentral.platform.metrics.labels.LabelValues;
+import com.ringcentral.platform.metrics.names.MetricName;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+import java.util.function.Function;
+
+import static com.ringcentral.platform.metrics.names.MetricName.name;
+
+/**
+ * An {@link ExecutorService} wrapper that provides the following metrics:
+ * <ul>
+ *   <li>{@link com.ringcentral.platform.metrics.rate.Rate} for submitted tasks, with the name {@code metricKeyProvider.apply(name("submitted"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.counter.Counter} for running tasks: {@code metricKeyProvider.apply(name("running"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.rate.Rate} for completed tasks: {@code metricKeyProvider.apply(name("completed"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.timer.Timer} for task idleness, measuring the time between task submission and the start of execution:
+ *       {@code metricKeyProvider.apply(name("idle"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.timer.Timer} for task execution time: {@code metricKeyProvider.apply(name("execution"))}</li>
+ * </ul>
+ *
+ * If the underlying (parent) {@link ExecutorService} is a {@link ForkJoinPool}, it additionally provides the following {@link com.ringcentral.platform.metrics.var.longVar.LongVar} metrics:
```

Comment:
Should we leave here information about `ThreadPoolExecutor`?

Reply:
Sure, I'll add the info.

Reply:
Added.